### PR TITLE
transaction - list external resources from cli

### PR DESCRIPTION
**Pull request by AlbsAgren**
added cli option --list-external-resources: lists id, alias and pid of known external resources

### CODE REVIEW

File: test/unittest/source/test_transaction.cpp [hunk 1]
```diff
@@ -372,6 +376,93 @@ namespace casual
...
+            -  connections: 
+               -  address: 127.0.0.1:7001
+)";
+
+            auto b = local::cli::domain( B);
+
+            auto a = local::cli::domain( A);
+            gateway::unittest::fetch::until( gateway::unittest::fetch::predicate::outbound::connected());
+
+            EXPECT_TRUE(administration::unittest::cli::command::execute( R"(casual transaction --list-external-resources --porcelain true)").string().empty());
```
Comment (lazan) [inline test/unittest/source/test_transaction.cpp:452]: nitpick: ws after `(`

File: middleware/transaction/source/manager/admin/cli.cpp [hunk 2]
```diff
@@ -323,6 +348,35 @@ namespace casual
                         
                      } // resources
 
+                     namespace external
+                     {
+                        auto option()
+                        {
+                           return argument::Option{
+                              []()
```
Comment (lazan) [inline middleware/transaction/source/manager/admin/cli.cpp:356]: nitpick: This, to me, feels like a strange format.

**Review by lazan — APPROVED** [review]
Looking good!

One thing. The unittest is as "far out" as we can come. I would prefer a unittest that use the model to verify that we have the correct state. 
We still need unittest for the cli though, so this might just be waste of time... 
If the logic was more complex and we needed more unittest, the model-way would be the right one, but for this cli-way only might be the best...

What do @ahhud and @albinsjolund think?

**Comment by albinsjolund** [comment]
> Looking good!
> 
> One thing. The unittest is as "far out" as we can come. I would prefer a unittest that use the model to verify that we have the correct state. We still need unittest for the cli though, so this might just be waste of time... If the logic was more complex and we needed more unittest, the model-way would be the right one, but for this cli-way only might be the best...
> 
> What do @ahhud and @albinsjolund think?

I agree. I think we should look at cli testing and model testing as two different things, so cli testing should not replace testing the model, even if it could work sometimes for simple models :)

I was also wondering if we should have all cli tests in one place? I have only seen them in _middleware/administration/unittest/source/cli_. Or can't we access the transaction stuff from there @lazan ?

**Review by albinsjolund — APPROVED** [review]

File: middleware/transaction/source/manager/admin/cli.cpp (hunk 2 above)
Comment (AlbsAgren) [reply middleware/transaction/source/manager/admin/cli.cpp:356]: Is this preferable?
 ```
return argument::Option{ []()
    {
       ...
 ```
 I followed the convention used in the file but I can refactor all ocurrences if we think its an improvement.

**Pull request merged**